llama_index/vector_stores/types.py

```diff
@@ -56,18 +57,23 @@ class VectorStoreQueryMode(str, Enum):
     LINEAR_REGRESSION = "linear_regression"
 
 
-@dataclass
-class ExactMatchFilter:
+class ExactMatchFilter(BaseModel):
     key: str
     value: Union[str, int, float]
 
 
-@dataclass
-class MetadataFilters:
+class MetadataFilters(BaseModel):
     # TODO: support more advanced expressions
     filters: List[ExactMatchFilter]
 
 
+class VectorStoreQuerySpec(BaseModel):
+    query: str
```

Comment:
what if top_k is a kwarg so that the LLM doesn't need to fill it in if it doesn't have to (might reduce risk of it picking a bad top k value?)

Reply:
revised to make this optional, and updated prompt to instruct to not input unless user query asks for document to retrieve, also clamp to max 10 documents by default.